codec-classes-quic/src/main/java/io/netty/incubator/codec/quic/BoringSSLCertificateCallback.java

```diff
@@ -77,24 +78,35 @@ final public class BoringSSLCertificateCallback {
         KEY_TYPES.put("DH_RSA", KEY_TYPE_DH_RSA);
     }
 
-    public static final Set<String> SUPPORTED_KEY_TYPES = new LinkedHashSet<>(
+    private static final Set<String> SUPPORTED_KEY_TYPES = Collections.unmodifiableSet(new LinkedHashSet<>(
             Arrays.asList(KEY_TYPE_RSA,
                     KEY_TYPE_DH_RSA,
                     KEY_TYPE_EC,
                     KEY_TYPE_EC_RSA,
-                    KEY_TYPE_EC_EC));
+                    KEY_TYPE_EC_EC)));
 
     // Directly returning this is safe as we never modify it within our JNI code.
     private static final long[] NO_KEY_MATERIAL_CLIENT_SIDE =  new long[] { 0, 0 };
 
     private final QuicheQuicSslEngineMap engineMap;
     private final X509ExtendedKeyManager keyManager;
     private final String password;
+    private Map<String, String> serverKeyTypeOverride;
+    private Set<String> clientKeyTypeOverride;
 
-    BoringSSLCertificateCallback(QuicheQuicSslEngineMap engineMap, @Nullable X509ExtendedKeyManager keyManager, String password) {
+    BoringSSLCertificateCallback(QuicheQuicSslEngineMap engineMap, @Nullable X509ExtendedKeyManager keyManager, String password, Map.Entry<SslContextOption<?>, Object>[] ctxOptions) {
         this.engineMap = engineMap;
         this.keyManager = keyManager;
         this.password = password;
+
+        for (Map.Entry<SslContextOption<?>, Object> ctxOpt : ctxOptions) {
```

Comment:
Do all this in the `SslContext` implementation and just pass the `Set`directly.

Reply:
Using the defaults when the option does not exist

Reply:
Sounds good. Working on it.